pandas/core/reshape.py

```diff
@@ -69,7 +89,8 @@ class _Unstacker(object):
     """
 
     def __init__(self, values, index, level=-1, value_columns=None,
-                 fill_value=None):
```

Comment:
so what I would do instead of this, is simply change values -> obj. Where obj is a Series/DataFrame. Then you don't need a constructor argument at all, and *if needed* you can always ``.values``. Though to be honest this should be completely changed. That coerces all dtypes to a common one.